feat: queue topic permission extensions

roleyfoley opened the pull request:
## Intent of Change
<!-- Delete all that do not apply                      -->
- New feature (non-breaking change which adds functionality)

## Description
<!--- Describe your changes in detail -->
- Allow for extending queue and topic policies through extensions
- Fixes testing for the lb component to ensure that the lb port is considered deployed. This was changed recently to only treat a target enabled if the actual port has been deployed. 
- Fixes an issue in the Lb component which was attempting to access a global variable instead of using the helper function

## Motivation and Context
<!--- Why make this change? Link to any existing issues here -->
Useful for integrating with services that require a specific policy applied or need to add additional security policies around how the resource can be used

## How Has This Been Tested?
<!--- Include details of your testing environment, official tests or other methods -->
Tested locally

## Related Changes
<!--- If anything not covered by the headings below, add here  -->

### Prerequisite PRs:
<!--- Add a checklist of items or leave the default of "None" -->
- None

### Dependent PRs:
<!--- Add a checklist of items or leave the default of "None" -->
- None

### Consumer Actions:
<!--- Add a checklist of items or leave the default of "None"
What changes must a consumer of this repository make in order to utilise it?
-->
- None

ml019 commented:
@roleyfoley there is some stuff about load balancers in this PR which seems unrelated to the description. If you want it included, please update the descriptions so I understand the intent.

roleyfoley commented:
> @roleyfoley there is some stuff about load balancers in this PR which seems unrelated to the description. If you want it included, please update the descriptions so I understand the intent.

Fixed up and updated description
